ha/core/event_analyzer/parser/parser.py

```diff
@@ -187,9 +187,10 @@ def parse_event(self, msg: str) -> HealthEvent:
             resource_id = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.RESOURCE_ID.value]
             event_type = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.RESOURCE_STATUS.value]
             specific_info = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.SPECIFIC_INFO.value]
-            if specific_info and specific_info["generation_id"] and source == HEALTH_EVENT_SOURCES.MONITOR.value:
-                generation_id = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.SPECIFIC_INFO.value]["generation_id"]
-                specific_info = {"generation_id": generation_id, "pod_restart": 0}
+            if resource_type == "node":
+                if specific_info and specific_info.get("generation_id") and source == HEALTH_EVENT_SOURCES.MONITOR.value:
+                    generation_id = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.SPECIFIC_INFO.value]["generation_id"]
```

Comment:
Line 189 is already giving specific_info.
So we can short this line as,
`generation_id = specific_info["generation_id"]`

Reply:
Right. Thanks. Done